dpctl/tensor/_array_api.py

```diff
@@ -196,22 +207,20 @@ def dtypes(self, *, device=None, kind=None):
                 a dictionary of the supported data types of the specified
                 ``kind``
         """
-        if device is None:
-            device = dpctl.select_default_device()
-        elif isinstance(device, dpt.Device):
-            device = device.sycl_device
+        device = _get_device_impl(device)
         _fp64 = device.has_aspect_fp64
         if kind is None:
             return {
                 key: val
                 for key, val in self._all_dtypes.items()
-                if (key != "float64" or _fp64)
+                if _fp64 or (key != "float64" and key != "complex128")
             }
         else:
             return {
                 key: val
                 for key, val in self._all_dtypes.items()
-                if (key != "float64" or _fp64) and _isdtype_impl(val, kind)
+                if (_fp64 or (key != "float64" and key != "complex128"))
```

Comment:
Perhaps it is useful to also check for fp16 aspect and filter "float16" out if it can be present.

Reply:
It will already not be present, as it is not mandated by array API, and array API strictly requires only data types present in array API be present in dtypes list ([see here](https://data-apis.org/array-api/latest/API_specification/generated/array_api.info.dtypes.html#array_api.info.dtypes))